orchestrate.sh: 'Prompt rejected by provider (oversize)' kills dispatch on dense briefs — claude -p subprocess argv limit

## Symptom

`orchestrate.sh define <prompt>` (and presumably other phases that dispatch via `claude -p` subprocess) fails with:

```
[ERROR] Agent claude-sonnet returned unusable output: Prompt rejected by provider (oversize)
```

The failure is exit-on-first-failure: the rejected dispatch kills the entire orchestrate.sh run before any other provider's perspective is gathered. Zero synthesis artifacts produced.

## Repro

```bash
# Brief is ~3.2KB raw text. Persona-wrap + role-wrap likely push the full
# `claude -p` argv past the subprocess command-line limit.
PROMPT=$(cat brief.txt)   # ~3.2KB
orchestrate.sh define "$PROMPT" --premium --verbose --timeout 1200
```

Brief content is technical context for a define phase — feature list, infrastructure description, constraints. Not a pathological edge case; representative of real define-phase usage.

## Observed

- 1st dispatch: agent ran ~19 minutes (with `OCTOPUS_AGENT_TIMEOUT=1200` env override applied locally to circumvent the hardcoded 180s timeouts in `run_agent_sync` callers).
- Provider rejected the prompt as oversize.
- orchestrate.sh exit-on-first-failure terminated the run before remaining providers' perspectives could be gathered.
- No partial artifacts written to `~/.claude-octopus/results/`.

## Environment

- octo plugin: v9.39.0
- Claude Code: v2.1.147
- Shell: Git Bash (MINGW64) on Windows 11
- Provider: claude (claude-sonnet role), via `claude -p` subprocess
- Allowlist: `claude gemini ollama` (codex disabled, expected behavior)

## Expected behavior (suggestions, pick one or some)

1. **File-based input fallback**: when dispatched prompt exceeds N KB (configurable), write to a temp file and pass `--file <path>` instead of inline argv.
2. **Graceful chunking**: split overly large briefs across multiple turns within a single agent dispatch.
3. **Hard error with size hint**: detect the size pre-dispatch and fail fast with \"brief exceeds N KB, reduce to ≤ N KB or split phase\" — saves 19 min of wasted runtime.
4. **Continue-on-failure for non-critical providers**: when one provider rejects, continue with others rather than exit-on-first-failure.

## Relation to existing issues

Distinct from #408 (synthesis-stage hardcoded 180s) — that's the synthesis phase. This is the per-agent dispatch input-size boundary.

Locally I've patched `run_agent_sync` in agent-sync.sh to honor `OCTOPUS_AGENT_TIMEOUT` unconditionally (the existing env var is dead code because callers bypass the dynamic-timeout path). Happy to PR if the design is acceptable.

---

CC environment: Claude Max (\$200/mo) OAuth, no API key. Multi-provider via `OCTO_ALLOWED_PROVIDERS=claude gemini ollama`.

--- a/scripts/lib/agent-sync.sh
+++ b/scripts/lib/agent-sync.sh
@@ -84,8 +84,13 @@ run_agent_sync() {
     local role="${4:-}"   # Optional role override
     local phase="${5:-}"  # Optional phase context
 
-    # v8.19.0: Dynamic timeout calculation (when caller uses default 120)
-    if [[ "$timeout_secs" -eq 120 ]]; then
+    # OCTOPUS_AGENT_TIMEOUT env var overrides all caller-hardcoded values.
+    # Without this, callers passing explicit values (e.g. 300, 600) bypass the
+    # dynamic path and the env var has no effect — making it dead code (#410).
+    if [[ -n "${OCTOPUS_AGENT_TIMEOUT:-}" && "${OCTOPUS_AGENT_TIMEOUT}" =~ ^[0-9]+$ ]]; then
+        timeout_secs="$OCTOPUS_AGENT_TIMEOUT"
+    elif [[ "$timeout_secs" -eq 120 ]]; then
+        # v8.19.0: Dynamic timeout calculation (when caller uses default 120)
         local task_type_for_timeout
         task_type_for_timeout=$(classify_task "$prompt" 2>/dev/null) || task_type_for_timeout="standard"
         timeout_secs=$(compute_dynamic_timeout "$task_type_for_timeout" "$prompt")
@@ -360,6 +365,16 @@ ${provider_ctx}"
         _sync_status="${_classification%%:*}"
         _sync_reason="${_classification#*:}"
         if [[ "$_sync_status" == "failed" ]]; then
+            # Oversize rejections are a provider-input-size mismatch, not a hard
+            # run failure. Return 0 with empty output so multi-provider dispatch
+            # loops continue to gather perspectives from remaining providers (#410).
+            if [[ "$_sync_reason" == *"oversize"* || "$_sync_reason" == *"Prompt rejected by provider"* ]]; then
+                log WARN "Agent $agent_type prompt rejected as oversized — skipping provider (reduce session context or lower OCTOPUS_CONTEXT_BUDGET)"
+                type write_agent_status >/dev/null 2>&1 && write_agent_status "$agent_type" "skipped" "$tokens_in" 0 "Prompt rejected by provider (oversize)" "$_elapsed_ms" "" "$role" || true
+                rm -f "$temp_err" "$temp_out"
+                echo ""
+                return 0
+            fi
             log ERROR "Agent $agent_type returned unusable output: $_sync_reason"
             type write_agent_status >/dev/null 2>&1 && write_agent_status "$agent_type" "failed" "$tokens_in" "$(octo_estimate_tokens_for_file "$temp_out" 2>/dev/null || echo 0)" "$_sync_reason" "$_elapsed_ms" "" "$role" || true
             rm -f "$temp_err" "$temp_out"

